x-pack/test/security_solution_endpoint/services/endpoint.ts

```diff
@@ -164,30 +198,35 @@ export class EndpointTestResources extends FtrService {
           },
         };
 
-    // If we have a specific number of endpoint hosts to check for, then use that number,
-    // else we just want to make sure the index has data, thus just having one in the index will do
-    const size = ids.length || 1;
-
-    await this.retry.waitFor('endpoint hosts', async () => {
-      try {
-        const searchResponse = await this.esClient.search({
-          index: metadataCurrentIndexPattern,
-          size,
-          body,
-          rest_total_hits_as_int: true,
-        });
+    await this.waitForIndex(ids, metadataCurrentIndexPattern, body, timeout);
+  }
 
-        return searchResponse.hits.total === size;
-      } catch (error) {
-        // We ignore 404's (index might not exist)
-        if (error instanceof errors.ResponseError && error.statusCode === 404) {
-          return false;
+  async waitForUnitedEndpoints(ids: string[] = [], timeout = this.config.get('timeouts.waitFor')) {
```

Comment:
If my comment above makes sense to you (about actualyl checking the united index rather than the other one), then maybe we don't need this method?

If you do keep it, please add JSDOCs that describe what this method does nad how its different from the one above.